### PR TITLE
loader: Use new eBPF ISA feature probes

### PR DESCRIPTION
**Pull request by pchaigno**
See commits for details.

### CODE REVIEW

**Comment by pchaigno** [comment]
/test-1.23-net-next

File: pkg/datapath/loader/compile.go [hunk 1]
```diff
@@ -163,18 +163,17 @@ func GetBPFCPU() string {
 			// added in the same release.
 			// We want to enable v3 only on kernels 5.10+ where we have
 			// tested it and need it to work around complexity issues.
-			if h := manager.GetHelpers("sched_cls"); h != nil {
+			h := manager.GetHelpers("sched_cls")
+			if manager.GetMisc().HaveV3ISAExtension && h != nil {
```
Comment (kkourt) [inline pkg/datapath/loader/compile.go:167]: Out of curiosity, was there a case where h != nil  but `HaveV3ISAExtension` was false?

**Review by kkourt — APPROVED** [review]
Thanks, Paul.

File: pkg/datapath/loader/compile.go (hunk 1 above)
Comment (pchaigno) [reply pkg/datapath/loader/compile.go:167]: Not unless there are some weird backports being performed e.g. on RHEL. I just wanted to make the condition explicit :-)

**Comment by pchaigno** [comment]
GKE failed with known flake https://github.com/cilium/cilium/issues/17307. Other tests are passing and Kornilios reviewed. Marking ready to merge.

**Review by qmonnet — APPROVED** [review]

**Pull request merged**